### PR TITLE
Added new audio device functions to set/get the audio device

### PR DESCRIPTION
**Pull request by marcelofg55**
This PR adds new functions for audio device handling:
```
	Array get_device_list();
	String get_device();
	void set_device(String device);
```
These can be called from GDscript using for example: `AudioServer.get_device_list()`

This simple project can be used to test this new functionality: [Device Changer.zip](https://github.com/godotengine/godot/files/1845675/Device.Changer.zip)

Implemented and tested: WASAPI, CoreAudio, PulseAudio and ALSA.

### CODE REVIEW

**Comment by marcelofg55** [comment]
I've just finished implementing this.
I had to re-write some parts of the PulseAudio driver to make it work, so it would be nice to give that some testing before merging. If the other drivers could also be tested it would be great as well.

File: drivers/pulseaudio/audio_driver_pulseaudio.cpp [hunk 1]
```diff
@@ -205,37 +176,101 @@ Error AudioDriverPulseAudio::init() {
...
-	if (pulse == NULL) {
-		fprintf(stderr, "PulseAudio ERR: %s\n", pa_strerror(error_code));
-		ERR_FAIL_COND_V(pulse == NULL, ERR_CANT_OPEN);
-	}
+	pa_str = pa_stream_new(pa_ctx, "Sound", &spec, NULL);
+	ERR_FAIL_COND_V(pa_ctx == NULL, ERR_CANT_OPEN);
+
+	const char *dev = device_name == "Default" ? NULL : device_name.utf8().get_data();
+	pa_stream_flags flags = pa_stream_flags(PA_STREAM_INTERPOLATE_TIMING | PA_STREAM_ADJUST_LATENCY | PA_STREAM_AUTO_TIMING_UPDATE);
+	int error_code = pa_stream_connect_playback(pa_str, dev, &attr, flags, NULL, NULL);
```
Comment (vnen) [inline drivers/pulseaudio/audio_driver_pulseaudio.cpp]: I'm getting a crash in here with a message: `Assertion 's' failed at pulse/stream.c`. Not sure what it's about.

**Comment by akien-mga** [comment]
Confirmed with the just-reenabled Travis iphone builds: https://travis-ci.org/godotengine/godot/jobs/364543876

File: drivers/pulseaudio/audio_driver_pulseaudio.cpp (hunk 1 above)
Comment (marcelofg55) [reply drivers/pulseaudio/audio_driver_pulseaudio.cpp]: That's with the default device? Crashes every time?

Comment (vnen) [reply drivers/pulseaudio/audio_driver_pulseaudio.cpp]: Yes, every time. Even the project manager.

Comment (marcelofg55) [reply drivers/pulseaudio/audio_driver_pulseaudio.cpp]: What distro/pulseaudio version?

**Pull request merged**